src/nnue/nnue_feature_transformer.h

```diff
@@ -420,78 +420,138 @@ namespace Stockfish::Eval::NNUE {
 
       // Now update the accumulators listed in states_to_update[], where the last element is a sentinel.
 #ifdef VECTOR
-      for (IndexType j = 0; j < HalfDimensions / TileHeight; ++j)
+
+      if (   states_to_update[1] == nullptr
```

Comment:
In the future it could be tried to apply it to any update and just special-case within the loop. These cases might be common enough to offset the branching cost.

Reply:
Yes agree we should try it in a subsequent PR.